com.unity.render-pipelines.core/Editor/Volume/VolumeComponentProvider.cs

```diff
@@ -47,35 +48,49 @@ public VolumeComponentProvider(VolumeProfile target, VolumeComponentListEditor t
             m_TargetEditor = targetEditor;
         }
 
-        public void CreateComponentTree(List<Element> tree)
+        static readonly Dictionary<Type, List<(string, Type)>> s_SupportedVolumeComponentsForRenderPipeline = new();
+
+        static List<(string, Type)> GetSupportedVolumeComponents(Type currentPipelineType)
         {
-            tree.Add(new GroupElement(0, "Volume Overrides"));
+            if (s_SupportedVolumeComponentsForRenderPipeline.TryGetValue(currentPipelineType,
+                out var supportedVolumeComponents))
+                return supportedVolumeComponents;
 
-            var types = VolumeManager.instance.baseComponentTypeArray;
-            var rootNode = new PathNode();
+            supportedVolumeComponents = FilterVolumeComponentTypes(
+                VolumeManager.instance.baseComponentTypeArray, currentPipelineType);
+            s_SupportedVolumeComponentsForRenderPipeline[currentPipelineType]= supportedVolumeComponents;
 
+            return supportedVolumeComponents;
+        }
+
+        static List<(string, Type)> FilterVolumeComponentTypes(Type[] types, Type currentPipelineType)
+        {
+            var volumes = new List<(string, Type)>();
             foreach (var t in types)
             {
-                // Skip components that have already been added to the volume
-                if (m_Target.Has(t))
-                    continue;
-
                 string path = string.Empty;
 
-                // Look for a VolumeComponentMenu attribute
                 var attrs = t.GetCustomAttributes(false);
 
                 bool skipComponent = false;
+
+                // Look for the attributes of this volume component and decide how is added and if it needs to be skipped
                 foreach (var attr in attrs)
                 {
-                    if (attr is VolumeComponentMenu attrMenu)
-                        path = attrMenu.menu;
-
-                    if (attr is HideInInspector attrHide)
-                        skipComponent = true;
-
-                    if (attr is ObsoleteAttribute attrDeprecated)
-                        skipComponent = true;
+                    switch (attr)
+                    {
+                        case VolumeComponentMenu attrMenu:
+                        {
+                            path = attrMenu.menu;
+                            if (attrMenu is VolumeComponentMenuForRenderPipeline supportedOn)
+                                skipComponent |= !supportedOn.pipelineTypes.Contains(currentPipelineType);
```

Comment:
Perhaps this should take into account inheritance similar to CustomEditorForRenderPipeline (CustomEditorAttributes.cs around line 70 in trunk). To handle cases where users inherit form URP but only make minor changes.

Reply:
I dont know if it should. The CustomEditorForRenderPipeline is not taking into account the full inheritance tree.